modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql

```diff
@@ -7137,6 +7137,16 @@ type Observation {
   """
   itc: Itc!
 
+  """
+  Enclosing group, if any.
+  """
+  groupId: GroupId
+
+  """
+  Index in enclosing group or at the top level if ungrouped. If left unspecified on creation, observation will be added last in its enclosing group or at the top level. Cannot be set to null.
+  """
+  groupIndex: NonNegShort
```

Comment:
Should this be `NonNegShort!` if it can't be null?

Reply:
Hmm, maybe? They are nullable in the database, but none of the rows on staging are.  I copied this over from another part of the schema.

<img width="431" alt="image" src="https://github.com/gemini-hlsw/lucuma-odb/assets/10114577/af4d1fd6-0a02-45cb-b977-cfb31e997d80">

<img width="155" alt="image" src="https://github.com/gemini-hlsw/lucuma-odb/assets/10114577/29c4263a-5789-47e2-916a-58f5cebebfc9">

It looks like `groupIndex` is nonnullable? @tpolecat ?

Reply:
Should be `NOT NULL`. It starts off `NULL` here https://github.com/gemini-hlsw/lucuma-odb/blob/main/modules/service/src/main/resources/db/migration/V0290__groups.sql#L39 but it's set to `NOT NULL` at line 57.

Reply:
```
lucuma-odb=# \d t_observation
                                                                                                                                                                                                                        Table "public.t_observation"
           Column           |            Type             | Collation | Nullable |                                                                                                                                                                                          Default                                                                                                                                                                                          
----------------------------+-----------------------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 c_program_id               | d_program_id                |           | not null | 
...
 c_group_id                 | d_group_id                  |           |          | 
 c_group_index              | smallint                    |           | not null | 
...
```

Reply:
Thanks, fixed (and in the other place it was used too)